kandora.bot/utils/TenhouParser/TenhouLogParserNew.cs

```diff
@@ -0,0 +1,119 @@
+﻿using kandora.bot.http;
+using kandora.bot.models;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace kandora.bot.utils.TenhouParser;
+
+public static class TenhouLogParserNew
+{
+    public static RiichiGame ParseTenhouFormatGame(string payload, GameType gameType)
+    {
+        var game = JsonSerializer.Deserialize<TenhouGame>(payload);
+        var node = JsonNode.Parse(payload);
+        var riichiGame = game.ToRiichiGame(node.SelectByPath("$.log"));
+        riichiGame.GameType = gameType;
+
+        return riichiGame;
+    }
+
+    public static RiichiGame ToRiichiGame(this TenhouGame tenhouGame, JsonNode resultGame)
+    {
+        return new RiichiGame()
+        {
+            Dan = tenhouGame.Dans,
+            Ref = tenhouGame.Reference,
+            Ver = tenhouGame.Version,
+            Rule = tenhouGame.Rule.ToRiichiRule(),
+            FinalRankDeltas = tenhouGame.Scores.GetRanks(),
+            FinalScores = tenhouGame.Scores.GetScores(),
+            Names = tenhouGame.Names,
+            UserIds = tenhouGame.GetIds(),
+            Rate = tenhouGame.Rate,
+            Lobby = tenhouGame.Lobby,
+            Sx = tenhouGame.Sx,
+            Ratingc = tenhouGame.Rating,
+            Title = tenhouGame.Title,
+            Rounds = GetRiichiRounds(resultGame.AsArray())
+        };
+    }
+
+    static int[] GetScores(this float[] sc)
+    {
+        var scores = new List<int>();
```

Comment:
Using linq utils would help readability and keep thing more compact :)
`return sc.Select(item => (int)item).ToArray();`